### PR TITLE
Felix config field for range of route tables to use

### PR DESCRIPTION
**Pull request by neiljerram**
## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: lib/validator/v3/validator_test.go [hunk 1]
```diff
@@ -610,6 +610,12 @@ func init() {
 		Entry("should reject an invalid BPFDataIfacePattern value '*'", api.FelixConfigurationSpec{BPFDataIfacePattern: "*"}, false),
 		Entry("should accept a valid BPFDataIfacePattern value 'eth.*'", api.FelixConfigurationSpec{BPFDataIfacePattern: "eth.*"}, true),
 
+		Entry("should accept valid route table range", api.FelixConfigurationSpec{RouteTableRange: &api.RouteTableRange{Min: 1, Max: 250}}, true),
+		Entry("should reject route table range min too small", api.FelixConfigurationSpec{RouteTableRange: &api.RouteTableRange{Min: 0, Max: 250}}, false),
+		Entry("should reject route table range min negative", api.FelixConfigurationSpec{RouteTableRange: &api.RouteTableRange{Min: -5, Max: 250}}, false),
+		Entry("should reject route table range max < min", api.FelixConfigurationSpec{RouteTableRange: &api.RouteTableRange{Min: 50, Max: 45}}, false),
+		Entry("should reject route table range max too large", api.FelixConfigurationSpec{RouteTableRange: &api.RouteTableRange{Min: 1, Max: 253}}, false),
+
```
Comment (song-jiang) [inline lib/validator/v3/validator_test.go:618]: Worth adding a {Min: 8, Max:8} test case.

Comment (neiljerram) [reply lib/validator/v3/validator_test.go:618]: Will do, thanks!

File: lib/validator/v3/validator.go [hunk 2]
```diff
@@ -1234,6 +1235,22 @@ func validateRuleMetadata(structLevel validator.StructLevel) {
 	validateObjectMetaAnnotations(structLevel, ruleMeta.Annotations)
 }
 
+func validateRouteTableRange(structLevel validator.StructLevel) {
+	r := structLevel.Current().Interface().(api.RouteTableRange)
+	if r.Min >= 1 && r.Max >= r.Min && r.Max <= 250 {
```
Comment (realgaurav) [inline lib/validator/v3/validator.go:1240]: I noticed [1-252] as valid table id values. Did we deliberately leave out 251,252?

Thanks

**Comment by caseydavenport** [comment]
Docs here: https://github.com/projectcalico/calico/pull/3477

**Pull request merged**